### ISSUE TITLE
getting an event when the audio file ended

### ISSUE THREAD

**Issue by lawsdontapplytopigs** [MartinSStewart/elm-audio#1]
i know in javascript you can do something like

```javascript
audio_node.addEventHandler("ended", console.log("audio ended"));
```
what would be the best way of having something like that in this package? I'm writing a music player and I need to know when a song ends so I can play the next one.

I started digging through the code and i saw that the duration in seconds is getting passed from JS to elm, but haven't dug enough in the elm code to see how I can get that info in my model.

I thought that if I had that info I could set a timer and change the song once that was done, but that feels a bit hacky to me.
The approach I would prefer would be to have some sort of subscription for it so it emits a Msg or something of that nature.

TL;DR: Using this package, what would be the best way to be noticed on the elm side when a song finished playing?
Any guidance would be appreciated

**Comment by MartinSStewart** [comment]
`I'm writing a music player and I need to know when a song ends so I can play the next one.`
Could you decide in advance when the songs will play?
```
audio : Model -> Audio
audio model = Audio.batch 
    [ Audio.audio model.song1 model.startTime
    , Audio.audio model.song2 (Duration.addTo song1Length model.startTime) 
    ]

-- Unfortunately for now, elm-audio lacks a way to get the length of a song so we'll need to hard code it or load that data from elsewhere
song1Length = Duration.seconds 200
```

That said, elm-audio currently isn't well suited for playing a large music collection. It doesn't support streaming music and can't unload songs from memory when they have finished playing. The normal <audio> tab might be better suited for this.

**Comment by lawsdontapplytopigs** [comment]
ahhh I see.

Yeah, I can't decide in advance when songs will play. I want the user to be able to arbitrarily change the song at any point in time, be able to seek forward and backward, etc. Think spotify's music playing capabilities.

Even so, I think this serves as a potential future improvement of the library.

Anyway, I'll just write my own javascript and do it all through ports, but thanks a lot for clarifying.

**Comment by MartinSStewart** [comment]
Jumping to other parts of a song and adjusting when the following songs should play is straight forward to do with elm-audio. In this example all you'd need to do to support seeking is change model.startTime so that the song continues playing from where the user wants to continue.
```elm
audio : Model -> Audio
audio model = 
    Audio.batch 
        [ Audio.audio model.song1 model.startTime
        , Audio.audio model.song2 (Duration.addTo model.startTime song1Length) 
        ]
```

`Anyway, I'll just write my own javascript and do it all through ports, but thanks a lot for clarifying.`
No problem! It's good to get feedback on what features are missing from elm-audio
